### PR TITLE
Feature/version tools work work

### PR DESCRIPTION
**Pull request by Kanti**

### CODE REVIEW

File: src/Process/StateChangeCallback.php [hunk 1]
```diff
@@ -3,9 +3,19 @@
...
 
-use Symfony\Component\Console\Output\ConsoleSectionOutput;
+use PHPSu\Tools\EnvironmentUtility;
 use Symfony\Component\Console\Output\OutputInterface;
 
+if (!class_exists('ConsoleSectionOutput', false)) {
+    if (version_compare((new EnvironmentUtility())->getSymfonyProcessVersion(), '4.0.0', '>=')) {
+        \class_alias(\Symfony\Component\Console\Output\ConsoleSectionOutput::class, 'ConsoleSectionOutput');
+    } else {
+        \class_alias(\PHPSu\Tools\ConsolePolyfill\ConsoleSectionOutput::class, 'ConsoleSectionOutput');
```
Comment (Kanti) [inline src/Process/StateChangeCallback.php]: maybe we could rename our class like the original. Than we could remove this if condition in every file. And only have this condition in the Polyfill File.

What do you think about this @ChrisB9?

**Comment by Kanti** [comment]
@ChrisB9 `90.85%` we are getting closer to our goal

File: composer.json [hunk 2]
```diff
@@ -39,7 +39,8 @@
   "autoload": {
     "psr-4": {
       "PHPSu\\": "src"
-    }
+    },
+    "files": ["src/polyfill.php"]
```
Comment (ChrisB9) [inline composer.json]: that's a cool idea

File: src/polyfill.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?php
+declare(strict_types=1);
+
+namespace PHPSu;
+
+use PHPSu\Tools\EnvironmentUtility;
+use Symfony\Component\Console\Output\ConsoleSectionOutput;
+
+if (!class_exists(ConsoleSectionOutput::class) && version_compare((new EnvironmentUtility())->getSymfonyConsoleVersion(), '4.0.0', '<')) {
```
Comment (ChrisB9) [inline src/polyfill.php]: :+1:

File: src/polyfill.php [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?php
```
Comment (ChrisB9) [inline src/polyfill.php]: can we just put this into a different directory? i personally dont like having something like this unimportant in the "root"-directory - what do you think?

File: tests/TestHelper/BufferedConsoleOutput.php [hunk 5]
```diff
@@ -0,0 +1,140 @@
+<?php
```
Comment (ChrisB9) [inline tests/TestHelper/BufferedConsoleOutput.php]: is that file copied? can you add the source?

File: src/Process/StateChangeCallback.php (hunk 1 above)
Comment (ChrisB9) [reply src/Process/StateChangeCallback.php]: That sounds like a great idea and i saw your implementation and liked it so far!

File: src/polyfill.php (hunk 4 above)
Comment (Kanti) [reply src/polyfill.php]: We could put our version of 
`\Symfony\Component\Console\Output\ConsoleSectionOutput` and this polyfill.php into the same directory.

like: `src/Polyfills/auto-polyfill.php`

Comment (ChrisB9) [reply src/polyfill.php]: splendid idea

File: src/Cli/InfoCliCommand.php [hunk 6]
```diff
@@ -34,7 +34,7 @@ public function printDependencyVersions(OutputInterface $output, SymfonyStyle $s
             ['Dependency', 'Installed', 'Version'],
             [
                 ['rsync', $environmentUtility->isRsyncInstalled() ? '✔' : '✘', $environmentUtility->getRsyncVersion()],
-                ['mysql-client', $environmentUtility->isMysqlDumpInstalled() ? '✔' : '✘', $environmentUtility->getMysqlDumpVersion()['mysqlVersion']],
+                ['mysql-client', $environmentUtility->isMysqlDumpInstalled() ? '✔' : '✘', $environmentUtility->getMysqlDumpVersion()['mysqlVersion']], // name mysql-client is misleading. The version gettet is the Server Version for that client
```
Comment (ChrisB9) [inline src/Cli/InfoCliCommand.php]: good point, we then should rename it to what it actually is to make sure there is no confusion that this is neither the "mysql-client-tool version" nor the "mysql-server version" or anything else but the "mysql-server version the client was built for"
- therefore i suggest renaming it to "mysql-distribution" (official name in the versioning) or "mysql-client-build-for", what do you think?

File: src/Command/DatabaseCommand.php [hunk 7]
```diff
@@ -262,4 +260,12 @@ private function excludeParts(string $database): string
         }
         return '';
     }
+
+    private function getCombinationPipe(string $targetDatabase): string
+    {
+        $escapedTargetDatabase = '`' . str_replace('`', '``', $targetDatabase) . '`';
+        $intermediateSql = sprintf('CREATE DATABASE IF NOT EXISTS %s;USE %s;', $escapedTargetDatabase, $escapedTargetDatabase);
+        $combinationPipe = ' | (echo ' . escapeshellarg($intermediateSql) . ' && cat) | ';
```
Comment (ChrisB9) [inline src/Command/DatabaseCommand.php]: `return ' | (echo ' . escapeshellarg($intermediateSql) . ' && cat) | ';`

File: src/Polyfills/auto-polyfill.php [hunk 8]
```diff
@@ -1,11 +1,13 @@
 <?php
 declare(strict_types=1);
 
-namespace PHPSu;
+namespace PHPSu\Polyfills;
 
 use PHPSu\Tools\EnvironmentUtility;
 use Symfony\Component\Console\Output\ConsoleSectionOutput;
 
+// @codeCoverageIgnoreStart
```
Comment (ChrisB9) [inline src/Polyfills/auto-polyfill.php]: i see you suffer from it too :(

Comment (Kanti) [reply src/Polyfills/auto-polyfill.php]: if this code fails our tests fail. so

File: src/Command/DatabaseCommand.php (hunk 7 above)
Comment (Kanti) [reply src/Command/DatabaseCommand.php]: I would like to merge this PR before we change anything else.

File: src/Cli/InfoCliCommand.php (hunk 6 above)
Comment (Kanti) [reply src/Cli/InfoCliCommand.php]: I will name it `mysql-distribution`. We still have to rework the info command anyway.

**Pull request merged**